Release/src/http/client/http_client_asio.cpp

```diff
@@ -1316,25 +1346,78 @@ class asio_context : public request_context, public std::enable_shared_from_this
         {
             // more data need to be read
             const auto this_request = shared_from_this();
-            writeBuffer.putn_nocopy(boost::asio::buffer_cast<const uint8_t *>(m_body_buf.data()),
-                             static_cast<size_t>(std::min(static_cast<uint64_t>(m_body_buf.size()), m_content_length - m_downloaded)))
```

Comment:
One think I'm not totally clear about is the difference between the result of this std::min evaluation and the subsequent:

`writtenSize = op.get();`

size. Could you elaborate a little on what's happening?